### ISSUE TITLE
Support for nullable type

### ISSUE THREAD

**Issue by drug007** [tamediadigital/asdf#72]
What is the way to have optional/nullable types like:
```
import std.typecons : Nullable;

struct Foo
{
    Nullable!int nullable_int;
}

auto foo = Foo();
assert (serialized(foo) == `{}`);

foo.nullable_int = 100;
assert (serialized(foo) == `{"nullable_int": 100}`);
```
?

**Comment by 9il** [comment]
`@serializationTransformIn!fin` and `@serializationTransformOut!fout` can be used where `fin`, fout is your custom functions for serialisation and deserialisation.

Also you may want to add PR to support `Nullable` out of the box.

**Comment by 9il** [comment]
Other approach is `@serializedAs` on top of you wrapper for Nullable.

**Comment by drug007** [comment]
I've tried to make PR, but can't realize why `Nullable` in some cases tries to use get on null payload (I've met this behavior before), namely in `serialization.d:1248` it tries get payload instead of getting the `Nullable` itself and fails in runtime.

**Comment by drug007** [comment]
Closed via #83
